mesonbuild/modules/python.py

```diff
@@ -159,32 +162,94 @@ def extension_module_method(self, args: T.Tuple[str, T.List[BuildTargetSource]],
 
             kwargs['install_dir'] = self._get_install_dir_impl(False, subdir)
 
+        target_suffix = self.suffix
+
+        limited_api_version = kwargs.pop('limited_api', '')
+        if not isinstance(limited_api_version, str):
+            raise InvalidArguments('"limited_api" argument must be a string.')
+
         new_deps = mesonlib.extract_as_list(kwargs, 'dependencies')
-        has_pydep = any(isinstance(dep, _PythonDependencyBase) for dep in new_deps)
-        if not has_pydep:
+        pydep = next((dep for dep in new_deps if isinstance(dep, _PythonDependencyBase)), None)
+        if pydep is None:
             pydep = self._dependency_method_impl({})
             if not pydep.found():
                 raise mesonlib.MesonException('Python dependency not found')
             new_deps.append(pydep)
             FeatureNew.single_use('python_installation.extension_module with implicit dependency on python',
                                   '0.63.0', self.subproject, 'use python_installation.dependency()',
                                   self.current_node)
+
+        if limited_api_version != '':
+
+            target_suffix = self.limited_api_suffix
+
+            limited_api_version_hex = self._convert_api_version_to_py_version_hex(limited_api_version)
+            if mesonlib.version_compare(limited_api_version, '<3.2'):
+                raise InvalidArguments(f'Limited API version invalid: {limited_api_version} (must be greater than 3.2)')
+            if mesonlib.version_compare(limited_api_version, '>' + pydep.version):
+                raise InvalidArguments(f'Limited API version too high: {limited_api_version} (detected {pydep.version})')
```

Comment:
Maybe prefixing `Python` in the exception messages would make things more clearer.

`Python limited API version too high: ...`